### PR TITLE
DPC-246: Refactor Blue Button client

### PR DESCRIPTION
**Pull request by nickrobison-usds**
**Why**
Per the changes required for DPC-25, we need a way to access the BlueButton backend from the AttributionService, in order to evaluate whether or not a patient has a previous claim that would result in an inferred attribution decision.

**What Changed**
Refactored the `BlueButtonClient` into its own module (`dpc-bluebutton`). This module is now injected into the required services by adding the `BlueButtonClientModule` to the list of modules in the DropwizardGuice injector.

The injection point is a `DropwizardAwareModule`, which means we can access the service configuration from within the module, as well as add our own configuration as an implemented Interface and get some type checking and type safety.

The BlueButtonClient also brings along its required health checks, which get injected into the host service automatically.

Configuration for the BBClient happens in the module now, but can be overridden within in the host service, if we so desire.

**Choices Made**
Originally, I wanted to make all of this an actual DropwizardModule, but that required using HK2 for service injection, which is started too late in the bootstrap process to be used in things like ManagedServices. In the end, it turned out to be much easier to make things a Guice module and keep the rest of our setup the same.

**Tickets closed**
DPC-246

**Future Work**
DPC-237: Remove FHIR client and use a more performance HTTP backend which can be shared amongst threads.

### CODE REVIEW

File: dpc-bluebutton/src/test/resources/test.application.conf [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    location = "../bbcerts/bb.keystore"
+  }
+
+  timeouts {
+    connectionTimeout: 5000 // ms
+    socketTimeout: 5000 // ms
+    requestTimeout: 5000 // ms
+  }
+
+  serverBaseUrl = "http://localhost:7788/v1/fhir/"
```
Comment (RickHawesUSDS) [inline dpc-bluebutton/src/test/resources/test.application.conf]: Is this a random port number or number that is necessary? If random, I suggest be close to a port number that we already use (eg. 8083) so we can write a port range that DPC talks on.

Comment (nickrobison-usds) [reply dpc-bluebutton/src/test/resources/test.application.conf]: It is just a random port, so I've made the change, good call!

File: dpc-bluebutton/src/main/java/gov/cms/dpc/bluebutton/client/DefaultBlueButtonClient.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package gov.cms.dpc.aggregation.bbclient;
+package gov.cms.dpc.bluebutton.client;
```
[the remark this answers is not in the document]
Comment (nickrobison-usds) [reply dpc-bluebutton/src/main/java/gov/cms/dpc/bluebutton/client/DefaultBlueButtonClient.java]: That's a good point, I renamed the class to `BlueButtonClientImpl` which is a Java standard for denoting the primary implementation of a given interface. Does that work?

File: dpc-aggregation/src/test/java/gov/cms/dpc/aggregation/AggregationMangerTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package gov.cms.dpc.aggregation;
```
[the remark this answers is not in the document]
Comment (nickrobison-usds) [reply dpc-aggregation/src/test/java/gov/cms/dpc/aggregation/AggregationMangerTest.java]: Fixed!

**Pull request merged**